### PR TITLE
:sunny: Added upvote and downvote function.

### PR DESCRIPTION
**Pull request by SalahAdDin**

### CODE REVIEW

File: app/templates/app/article_detail.html [hunk 1]
```diff
@@ -9,10 +9,34 @@
...
+                <button type="button">{{ article.downvotes }}</button>
+            </li>
         </ul>
     </div>
 {% endblock content %}
+{% block extra_js %}
+<script type="text/javascript">
+$(document).ready(function () {
+    $('div.vote-buttons li.vote-up').click(function (){
+        $.get("{% url 'article_detail' article.slug %}upvote", function(response) {
```
Comment (camilonova) [inline app/templates/app/article_detail.html]: url name should be `article_upvote`

File: app/templates/app/article_detail.html [hunk 2]
```diff
@@ -9,10 +9,34 @@
...
+            $('li.vote-up').addClass('selected');
+            $('li.vote-down button').prop('disabled', true);
+        });
+        return false;
+    });
+    $('div.vote-buttons li.vote-down').click(function (){
+        $.get("{% url 'article_detail' article.slug %}downvote", function(response) {
```
Comment (camilonova) [inline app/templates/app/article_detail.html]: url name should be `article_downvote`

File: app/views.py [hunk 3]
```diff
@@ -44,3 +47,45 @@ def get_context_data(self, **kwargs):
...
+    model = Article
+
+    def get(self, request, *args, **kwargs):
+        article = self.get_object()
+
+        if not request.session.get('article_id', article.id):
+            request.session['article_id'] = article.id
+            article.upvotes = F('upvotes') + 1
+            article.save()
+        # article = self.get_object()
```
Comment (camilonova) [inline app/views.py]: 😢

File: app/views.py [hunk 4]
```diff
@@ -44,3 +47,45 @@ def get_context_data(self, **kwargs):
...
+class ArticleUpVoteView(DetailView):
+
+    model = Article
+
+    def get(self, request, *args, **kwargs):
+        article = self.get_object()
+
+        if not request.session.get('article_id', article.id):
+            request.session['article_id'] = article.id
+            article.upvotes = F('upvotes') + 1
```
Comment (camilonova) [inline app/views.py:81]: 🏆  well done

File: app/views.py [hunk 5]
```diff
@@ -44,3 +47,45 @@ def get_context_data(self, **kwargs):
...
+    model = Article
+
+    def get(self, request, *args, **kwargs):
+        article = self.get_object()
+
+        if not request.session.get('article_id', article.id):
+            request.session['article_id'] = article.id
+            article.upvotes = F('downvotes') + 1
+            article.save()
+        # article = self.get_object()
```
Comment (camilonova) [inline app/views.py]: 😭

File: app/templates/app/article_detail.html [hunk 6]
```diff
@@ -9,10 +9,34 @@
...
+{% block extra_js %}
+<script type="text/javascript">
+$(document).ready(function () {
+    $('div.vote-buttons li.vote-up').click(function (){
+        $.get("{% url 'article_upvote' article.slug %}", function(response) {
+            $('li.vote-up').addClass('selected');
+            $('li.vote-down button').prop('disabled', true);
+        });
+        return false;
+    });
```
Comment (PabloVallejo) [inline app/templates/app/article_detail.html:39]: Add space after

**Pull request merged**